modules/claims_api/app/services/claims_api/disability_compensation/disability_document_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  module DisabilityCompensation
+    class DisabilityDocumentService < ServiceBase
+      LOG_TAG = '526_v2_Disability_Document_service'
+
+      def create_upload(claim:, pdf_path:, doc_type: 'L122', action: 'post', original_filename: nil)
+        unless File.exist? pdf_path
+          ClaimsApi::Logger.log('benefits_documents', detail: "Error creating upload doc: #{file_path} doesn't exist,
+                                                      claim_id: #{claim.id}")
+          raise Errno::ENOENT, pdf_path
+        end
+
+        body = generate_upload_body(claim:, doc_type:, pdf_path:, action:, original_filename:)
+        ClaimsApi::BD.new.upload_document(claim_id: claim.id, doc_type:, body:)
+      end
+
+      private
+
+      ##
+      # Generate form body to upload a document
+      #
+      # @return {parameters, file}
+      def generate_upload_body(claim:, doc_type:, pdf_path:, action:, original_filename: nil)
```

Comment:
I'm concerned this method will basically need to be duplicated in the other service files (if I am seeing the direction correctly.

Wondering if there is a way to move this to the service_base perhaps and make it universal enough that we can invoke this one method from the different service files? 

I just feel like this method should still be shared, acknowledging of course that the direction may become clearer with another service in the mix to see.

Reply:
good callout here. just had a sync with Matt and I and we were in agreeance 100%. I like the idea of an intermediary service, somewhat generic, and specific to the document functionality. The original idea was to move to servicebase but with that service, it handles most the root level and generic functionalities. The thought now is to have a DocumentService < ServiceBase that would handle some of these shared, generic document specific methods, and then we'll just have a Service for each form type, hopefully with just a singular "extraction" method to get the values needed to call the rest of the services within the DocumentService, which will be the builders. 

Hopefully that makes sense.